examples/ecs/send_and_receive_events.rs

```diff
@@ -0,0 +1,174 @@
+//! From time-to-time, you may find that you want to both send and receive and event of the same type in a single system.
+//!
+//! Of course, this results in an error: the borrows of [`EventWriter`] and [`EventReader`] overlap,
+//! if and only if the [`Event`] type is the same.
+//! One system parameter borrows the [`Events`] resource mutably, and another system parameter borrows the [`Events`] resource immutably immutably.
+//! If Bevy allowed this, there would be Undefined Behavior!
```

Comment:
Maybe this could say something like "this would violate Rust's mutability rules" instead - "UB" can be jargon